units/tests/api.rs

```diff
@@ -191,33 +191,19 @@ fn api_can_use_all_types_from_module_weight() {
 // `Debug` representation is never empty (C-DEBUG-NONEMPTY).
 #[test]
 fn api_all_non_error_types_have_non_empty_debug() {
+    macro_rules! check_debug {
+        ($t:tt, $field:tt; $($sub_field:tt),* $(,)?) => {
+            $(
+                let debug = format!("{:?}", $t.$field.$sub_field);
```

Comment:
Why is it checking fields rather than the type?

Also looks like this is the kind of test that should use `arbitrary`.

Reply:
Its checking that the debug output is not empty. The structs are artificial, just to hold all the types. So the fields (sub field) in this instance _is_ the type we are wanting to test.

Reply:
> Also looks like this is the kind of test that should use `arbitrary`.

That I can get behind. I did it like that yesterday in the `primitives` version of this file.

Reply:
But why not test all types one-by-one?

Reply:
During original development I just thought it was easier to have a constructor but IIRC it only ends up being used in this test. I'll take another look.